pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -291,6 +292,12 @@ func buildHTTPVirtualServices(
 			vs.Route = route
 		}
 
+		tr, err := createTimeoutRequest(*r.Timeouts)
+		if err != nil {
+			reportError(err)
+		}
+		vs.Timeout = tr.Timeout
```

Comment:
I think this should just be something like this?
```suggestion
		if r.Timeouts != nil && r.Timeouts.Request != nil {
			vs.Timeout = durationpb.New(r.Timeouts.Request.Duration)
		}
```

Reply:
@frankbu do we still need `createTimeoutRequest`  function to separate in future from `workloadRequest` timeout as per https://github.com/kubernetes-sigs/gateway-api/pull/1997#discussion_r1189088248 ?

Reply:
No. I think you're confusing istio HTTPRoute with gateway-api HTTPRoute. We've already split timeout out of gateway-api HTTPRoute as suggested in the comment (previously I had `spec.rules.timeout`, but now it is `spec.rules.timeouts.request`). If you look at the latest [GEP-1742 PR](https://github.com/kubernetes-sigs/gateway-api/pull/1997) you'll see that I've also added `backendRequest` (renamed from `workloadRequest`) under `spec.rules.timeouts`.

Reply:
If you would like to pencil in the code for `backendRequest`, I think the above code would be something like this:
```
		if r.Timeouts != nil {
			if r.Timeouts.Request != nil {
				vs.Timeout = durationpb.New(r.Timeouts.Request.Duration)
			}
			// TODO
			// if r.Timeouts.BackendRequest != nil {
				// if vs.Retries == nil {
				//	vs.Retries = create new one
				// }
				// vs.Retries.PerTryTimeout = durationpb.New(r.Timeouts.BackendRequest.Duration)
			// }
		}
```